test/currencyConvertor.ts

```diff
@@ -234,14 +266,93 @@ describe("CurrencyConvertor", () => {
         '100000',
         starkKeyToUint256('050e0343dc2c0c00aa13f584a31db64524e98b7ff11cd2e07c2f074440821f99'),
         '22', // positionId
-        ethers.utils.getAddress(zeroExTransaction.to),
+        zeroExTransaction.to,
         zeroExTransaction.data,
       )).to.be.revertedWith('Received USDC is less than minUsdcAmount');
     });
   });
+
+  describe("deposit ETH", async () => {
+    it("deposit ETH as USDC to Starkware", async () => {
+      const zeroExTransaction = await zeroExRequestEth('1000');
+
+      // get old balances
+      const userETHBalance: BigNumber = await signer.getBalance();
+      const starkwareUsdcBalance: BigNumber = await usdcTokenContract.balanceOf(
+        starkwareContractAddress,
+      )
+
+      const tx = await currencyConvertor.depositEth(
+        '1000',
+        starkKeyToUint256('050e0343dc2c0c00aa13f584a31db64524e98b7ff11cd2e07c2f074440821f99'),
+        '22', // positionId
+        zeroExTransaction.to,
+        zeroExTransaction.data,
+        { value: zeroExTransaction.value },
+      );
+
+      const blocks = await tx.wait();
+      const events = _.chain(blocks.events!)
+      .filter((e) => e.event === 'LogConvertedDeposit')
+      .value();
+
+      const event = events[0];
+      expect(event.args?.tokenFromAmount.toString()).to.equal(zeroExTransaction.value);
+      // expect(event.args?.tokenFrom.toLowerCase()).to.equal('eth');
+
+      // get new balances
+      const newUserETHBalance: BigNumber = await signer.getBalance();
+      const newStarkwareUsdcBalance: BigNumber = await usdcTokenContract.balanceOf(
+        starkwareContractAddress,
+      );
+
+      expect(newUserETHBalance.lt(userETHBalance)).to.be.true;
+      expect(newStarkwareUsdcBalance.gt(starkwareUsdcBalance)).to.be.true;
+    });
+
+    it("deposit ETH to USDC without enough funds", async () => {
+      const zeroExTransaction = await zeroExRequestEth('1000');
+
+      await expect(currencyConvertor.depositEth(
+        '1',
+        starkKeyToUint256('050e0343dc2c0c00aa13f584a31db64524e98b7ff11cd2e07c2f074440821f99'),
+        '22', // positionId
+        zeroExTransaction.to,
+        zeroExTransaction.data,
+        { value: '1' },
+      )).to.be.reverted;
+    });
+
+    it("deposit ETH to USDC with too small of swap", async () => {
+      const zeroExTransaction = await zeroExRequestEth('1000');
+
+      await expect(currencyConvertor.depositEth(
+        '100000000000',
+        starkKeyToUint256('050e0343dc2c0c00aa13f584a31db64524e98b7ff11cd2e07c2f074440821f99'),
+        '22', // positionId
+        zeroExTransaction.to,
+        zeroExTransaction.data,
+        { value: zeroExTransaction.value },
+      )).to.be.reverted; //  �y� % UniswapV2: INSUFFICIENT_OUTPUT_AMOUNT
```

Comment:
Can you add a revert string check to all these? E.g. just use `'UniswapV2: INSUFFICIENT_OUTPUT_AMOUNT'` and it will look for that as a substring